### ISSUE TITLE
Feedback & Feature Requests

### ISSUE THREAD

**Issue by sibalzer** [sibalzer/impfbot#5]
Feature Requests

To avoid cluttering the issue tracker with feature requests, please comment any requests here and we'll keep a list.

When available, I've linked a related issue or comment to add context to the request.

**Comment by glutamat33** [comment]
hey! did you get my twitter DMs?

**Comment by sibalzer** [comment]
Had to strict privacy setting turned. Should be working now.

**Comment by Wyall** [comment]
Does it matter if the birth date is written as 06 or 6 e.g. for the month of June? You changed it in the Readme but not everywhere

**Comment by sibalzer** [comment]
Doesn't matter. Only the year must be a 4 digt number.

**Comment by thetha** [comment]
Did someone manage to get a free slot with this yet?

**Comment by glutamat33** [comment]
I found 1 free slot today after waiting 2 days. 
Unfortunately I was too slow, it was already gone as I locked in...

**Comment by thetha** [comment]
I have started right now. Running 4 Hours, nothing yet.

**Comment by sibalzer** [comment]
Unfortunately, the policy according to which the vaccination centers publish the dates is not uniform and consistent, it is difficult to estimate when the dates are published (here in OS, most slots are published on Thu/Fri at each full hour from 11-18). To get single appointments in cities is almost impossible. The best chance is when batches > 5 are published. But you can also set the bot more agressive. Default is very passive. Just adjust ADVANCED sleep_between_requests_in_s & co (atm the bot waits 150s eg 2:30min between each successful request, 60s should be fine to. At 30s i got IP-Banned [Just reset the router to remove the Ban an get a new IP]).

**Comment by paulypeter** [comment]
@sibalzer I'd like to continue working on this bot, do you have something I should do next?

**Comment by sibalzer** [comment]
Im out of ideas atm. Maybe color coded logging? Discord notifications?

**Comment by paulypeter** [comment]
Alright, maybe I'll look into those!

**Comment by sibalzer** [comment]
Also  #39

**Comment by ToHindahl** [comment]
Otherwise, I have other ideas such as proxy support. That means a proxy.txt in which several proxies are inside and so you could make the intervals smaller and check the availability more often. Or I don't know to what extent it is possible to also state the available date in the notification?

**Comment by riher** [comment]
impfbot got me a vaccination today! :tada: 
Thanks a lot for your work! :+1:

**Comment by glutamat33** [comment]
Me too!
THANKS A LOT!!

**Comment by Tubbyx** [comment]
imfbot found a free slot but i was too slow. They must be releasing a lot of free slots today

**Comment by sibalzer** [comment]
> proxy support

Will be added

> what extent it is possible to also state the available date in the notification

Should also be possible i think

**Comment by thetha** [comment]
It worked!

**Comment by dtegunov** [comment]
Thank you so much for releasing this tool!

Each center is assigned one vaccine type in the response – vector or mRNA. Do you know if this is a hard constraint, or do they usually offer both types regardless? Depending on age and current Stiko guidelines, some may spend time watching a center with the wrong vaccine type instead of watching a more distant center with the right type. Would it be worth pointing this out and having a list of all centers and their advertised vaccine type in the readme?

**Comment by sibalzer** [comment]
Usually, the vaccination centers offer both types. 

Monitoring a more distant vaccination center would make no sense since you are bound to your local center in lower saxony (as far as i know). The type (mRNA/vector) is purely dependent on the age, so you can't monitor a center with the wrong type. The allocation is done by the backend of the vaccination portal. So the practical use of a list in the readme would be close to 0.

Eg:
27472 or 27474 and over 60 years old: IZ Cuxhaven - AstraZeneca or IZ Cuxhaven - Johnson & Johnson
27472 or 27474 and under 60 years old: IZ Cuxhaven - Moderna or IZ Cuxhaven - BioNtech

tldr: There is a 1-n relationship between vaccination centers, depending on age and zip code, allocated by the backend of the vaccination portal.

**Comment by paulypeter** [comment]
> Usually, the vaccination centers offer both types.

They do. During my appointment, there were people being vaccinated with AZ, some with BioNtech/Pfizer, and some with Moderna.

> you are bound to your local center in lower saxony

That is correct. This is implemented by the gov to reduce 'vaccination tourism'.

**Comment by paulypeter** [comment]
@sibalzer Are there any more issues on the roadmap for v3.0.0?

e. g. #58 or #59 ?

Otherwise, I think that's quite a big update already, so those could be implemented later?

**Comment by sibalzer** [comment]
v3.0.0 is big indeed. Lets schedul them for an later update.

**Comment by ronizzel** [comment]
This morning the bot alerted me for a free slot, and I successfully took the chance and am extremely happy I got my Erstimpfung already.
I'm very grateful for your work and hope, the impfbot can heal many people from "when is there a slot finally?"-anxiety :-)

**Comment by DennisWilken** [comment]
First of all, a big thank you for your work. I really appreciate it.  Today, I got three notifications with 1/109/1 slots. I didn't have luck though. 

I would like to run the group bot in parallel on the same Telegram channel. I haven't checked the code. Are there different messages for the two modi? It could be helpful to add something like "Einzeltermin" and "Gruppentermin" to the output message to clarify what people should click.

Another feature that I would like to see, is a notification mail in case there is an IP ban. There should be an option to send this mail to a different set of addresses to avoid spamming the group.

**Comment by paulypeter** [comment]
Currently, the bot only checks either for single or (exclusive or) group appointments. It then (currently) sends the same message.

A possible, albeit hopefully temporary, solution for you would be to run two bots, one configured for single appointments and one configured for group appointments.

@sibalzer We could include that in a later version, maybe?

**Comment by sibalzer** [comment]
> I would like to run the group bot in parallel on the same Telegram channel. I haven't checked the code. Are there different messages for the two modi? It could be helpful to add something like "Einzeltermin" and "Gruppentermin" to the output message to clarify what people should click.

We could add some custom prefix (in config) to the message eg:
[My Group bot for 49123] Freier Impfslot (11)! AstraZeneca/Vector

> Another feature that I would like to see, is a notification mail in case there is an IP ban. There should be an option to send this mail to a different set of addresses to avoid spamming the group.

We could indeed add a optional admin notification receiver list mail. (#94) 

> A possible, albeit hopefully temporary, solution for you would be to run two bots, one configured for single appointments and one configured for group appointments.

Dockerize everything! One problem is, we don't know if they even publish any group appoinments (at least I didn't find any during testing). The other is, that we would need some more advanced scheduling mechanism as sleep (apscheduler?). Because of that, I classify this as a very low priority.

**Comment by Doomnometron** [comment]
Ich bekomme die Mail Benachrichtigung nicht eingerichtet. Habe schon unterschiedliche Anbieter (web.de / GMail) probiert, aber keiner funktioniert.

web.de
smtp.web.de 587/tcp

> [ERROR] Couldn't send mail: SMTP AUTH extension not supported by server.

GMail
smtp.gmail.com 465/tcp

> 2021-06-09 18:21:56 [ERROR] Couldn't send mail: (535, b'5.7.8 Username and Password not accepted. Learn more at\n5.7.8 https://support.google.com/mail/?p=BadCredentials u18sm248744wmj.15 - gsmtp')

:-(

**Comment by paulypeter** [comment]
@Doomnometron 

Für GMail muss zunächst ein App-Passwort erstellt werden. Dieses kann man dann für den Impfbot verwenden. Hinweise dazu gibt es [hier](https://support.google.com/accounts/answer/185833?hl=de).

**Comment by Doomnometron** [comment]
@paulypeter 
Bei Google kann ich erst ein App Passwort einrichten, wenn ich 2FA aktiviert habe - habe ich aber nicht (ist kein aktiv genutztes Konto).

**Comment by paulypeter** [comment]
@Doomnometron 

Alles klar, diese Option habe ich auch gerade gefunden.

**Comment by sibalzer** [comment]
Ich hab grad mein altes Gmail Konto ausgegraben und eine Anleitung unter [docs/SETUP_EMAIL_SERVICES.md ](docs/SETUP_EMAIL_SERVICES.md ) geschrieben. Web.de konnte ich nicht testen sollte aber so passen.

**Comment by krohn** [comment]
It works. 
Got my slot after running the script for less than a week. 🎉
The bot alerted me some more times the other days, but I was too slow filling the forms.
Thanks a lot for your work! 👍

**Comment by quickcoffee** [comment]
Thanks for the work!
Just managed to book an appointment after I was too slow with the forms in the previous days, but left the tab with the calendar view open and switched between the months immediately after I got a notification from the impfbot!
😍

**Comment by AnjaMirenska** [comment]
Hi, how do I know whether the bot works? When I double-click on `windows_validate.bat`, nothing happens (well, for a very short moment I see in my task bar that apparently a window is open, but it disappears immediately). It doesn't matter whether I execute the file as administrator or not. Should I get a Telegram message in my bot in case the validation was successful?

**Comment by andy71** [comment]
Try to start the windows_validate.bat from a command shell. To open a command shell, please press Win+r and in this dialogue enter cmd and press enter.
In the command shell change the directory to where the windows_validate.bat is. You can use something like this...

`cd /D x:/path/to/windows_validate.bat`

Now try to start the windows_validate.bat. I am pretty sure you will see some more information which can help to find the issue.

**Comment by AnjaMirenska** [comment]
> Try to start the windows_validate.bat from a command shell. To open a command shell, please press Win+r and in this dialogue enter cmd and press enter.
> In the command shell change the directory to where the windows_validate.bat is. You can use something like this...
> 
> `cd /D x:/path/to/windows_validate.bat`
> 
> Now try to start the windows_validate.bat. I am pretty sure you will see some more information which can help to find the issue.

Ah, ok, thanks... I forgot that I haven't installed Python on my new laptop, this seems to be the problem. Thanks a lot again!
